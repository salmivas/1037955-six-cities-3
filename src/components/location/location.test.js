import React from "react";
import renderer from "react-test-renderer";
import Location from "./location.jsx";

const city = {
  id: 136,
  name: `Paris`,
  position: [48.8566, 2.3522],
  offersCount: 100,
};

const currentCity = {
  name: `Paris`,
};

const newCity = {
  name: `Cologne`,
};

describe(`Location should be rendered correctly`, () => {
  it(`<Location/> should render an active city name `, () => {
    const tree = renderer.create(
        <Location
          onLocationItemLinkClick = {() => {}}
          currentCity = {currentCity}
          city = {city}
        />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`<Location/> should render a city name`, () => {
    const tree = renderer.create(
        <Location
          onLocationItemLinkClick = {() => {}}
          currentCity = {newCity}
          city = {city}
        />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
