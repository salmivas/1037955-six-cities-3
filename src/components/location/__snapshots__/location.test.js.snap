// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location should be rendered correctly <Location/> should render a city name 1`] = `
<li
  className="locations__item"
>
  <a
    className="locations__item-link tabs__item"
    href="#"
    onClick={[Function]}
  >
    <span>
      Paris
    </span>
  </a>
</li>
`;

exports[`Location should be rendered correctly <Location/> should render an active city name  1`] = `
<li
  className="locations__item"
>
  <a
    className="locations__item-link tabs__item tabs__item--active"
    href="#"
    onClick={[Function]}
  >
    <span>
      Paris
    </span>
  </a>
</li>
`;
